File: CommonLib/src/main/java/com/enzo/commonlib/base/IBaseActivity.java
package com.enzo.commonlib.base;

import android.os.Bundle;

/**
 * 文 件 名: IBaseActivity
 * 创 建 人: xiaofangyin
 * 创建日期: 2017/11/28
 * 邮   箱: xiaofangyinwork@163.com
 */
public interface IBaseActivity {

    int getStatusBarColor();

    int getLayoutId();

    void initHeader();

    void initView();

    void initData(Bundle savedInstanceState);

    void initListener();
}

File: commonlibkt/src/main/java/com/enzo/commonlibkt/base/IBaseActivity.kt
package com.enzo.commonlibkt.base

import android.os.Bundle

/**
 * 文 件 名: IBaseActivity
 * 创 建 人: xiaofy
 * 创建日期: 2019/12/28
 * 邮   箱: xiaofangyinwork@163.com
 */
interface IBaseActivity {

    fun getLayoutId(): Int

    fun initHeader()

    fun initView()

    fun initData(savedInstanceState: Bundle?)

    fun initListener()
}